packages/components/src/font-size-picker/index.tsx

```diff
@@ -94,28 +82,16 @@ const UnforwardedFontSizePicker = (
 		availableUnits: [ 'px', 'em', 'rem' ],
 	} );
 
-	/**
-	 * The main font size UI displays a toggle group when the presets are less
-	 * than six and a select control when they are more.
-	 */
-	const fontSizesContainComplexValues = fontSizes.some(
-		( { size: sizeArg } ) => ! isSimpleCssValue( sizeArg )
-	);
 	const shouldUseSelectControl = fontSizes.length > 5;
-	const options = useMemo(
-		() =>
-			getFontSizeOptions(
-				shouldUseSelectControl,
-				fontSizes,
-				disableCustomFontSizes
-			),
-		[ shouldUseSelectControl, fontSizes, disableCustomFontSizes ]
+	const selectedFontSize = fontSizes.find(
+		( fontSize ) => fontSize.size === value
```

Comment:
That logic only ever had any effect when `shouldUseSelectControl` was `true` and the `CustomSelectControl` was rendered, because `ToggleGroupControl` used `value` and not `selectedOption.size`. I therefore moved this logic to `FontSizePickerSelect` so that it is near to where it's needed and not executed unnecessarily. 

https://github.com/WordPress/gutenberg/blob/bbf17a3f5c90f488bbb9b6b34146d5b3cadca859/packages/components/src/font-size-picker/select.tsx#L42-L43

Reply:
Thank you for the explanation! This makes sense for the fallback to `customOption`, but I'm still confused about where this piece of logic is implemented in this new version: 

https://github.com/WordPress/gutenberg/blob/d20cc8ba7e84d2385b1e257ac8ef397a515a66a8/packages/components/src/font-size-picker/utils.ts#L147-L149

Reply:
Maybe I get it — when `value` is `undefined`, this code will actually select the default option (since it also have a value of `undefined`):


https://github.com/WordPress/gutenberg/blob/bbf17a3f5c90f488bbb9b6b34146d5b3cadca859/packages/components/src/font-size-picker/select.tsx#L42-L43

If that's the reason, I must admit that the logic here is much less explicit than it used to be, and I would argue that it would greatly benefit from a unit test to make sure that we don't introduce a regression in the future without realizing.

Reply:
I also realised that the header hint is `default` on `trunk`, but it's an empty string in this PR.

Reply:
> Maybe I get it — when `value` is `undefined`, this code will actually select the default option (since it also have a value of `undefined`):

Yes that's right. I'll make the `find()` code more explicit with a ternary.

> I also realised that when the `value` prop is `undefined` the header hint:

👍 I've fixed this and restored most of the changes I made to `headerHint`. Got a bit ahead of myself there, I'll wait for https://github.com/WordPress/gutenberg/pull/44966 which has more substantial changes to the hint text.

Reply:
> Yes that's right. I'll make the find() code more explicit with a ternary.

Looking more into this, I found a couple more more differences with `trunk`

1. With more than 5 font sizes and when passing `value=""` to `FontSizePicker`, prior to your latest change that introduces the ternary, the `selectedOption` would have been `CUSTOM_OPTION` instead of `DEFAULT_OPTION` — that's because the empty string wouldn't match against the `undefined` value of the default option. This has been fixed by using the `value ?` ternary, but I though I'd flag it nonetheless
2. One difference that still remains between this PR and `trunk` is that, when `value` is `""`:
   - on `trunk`, the component shows a header hint that says "size default" and:
     - with 5+ options, the `CustomSelectControl` with the default option selected (and 
     - with less than 5 options, the `ToggleGroupControl` without no options selected 
   - in this PR, the component shows an empty custom value control input and the header hint says "size (custom)"


After fixing those discrepancies, I also suggest that we add unit tests to protect against these regressions in the future!

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/44891/commits/639db98a1e4bacd081de03609b57524f9fbb8507